src/Store/MongoDB/Store.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Store\MongoDB;
+
+use MongoDB\Client;
+use PhpLlm\LlmChain\Document\Document;
+use PhpLlm\LlmChain\Document\Metadata;
+use PhpLlm\LlmChain\Document\Vector;
+use PhpLlm\LlmChain\Store\VectorStoreInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @see https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/
+ *
+ * For this store you need t create a separate MongoDB Atlas Search index.
+ * The index needs to be created with the following settings:
+ * {
+ *     "fields": [
+ *         {
+ *             "numDimensions": 1536,
+ *             "path": "vector",
+ *             "similarity": "euclidean",
+ *             "type": "vector"
+ *         }
+ *     ]
+ * }
+ *
+ * Note, that the `path` needs to match the $vectorFieldName.
+ * For the similarity you can choose between `euclidean`, `cosine` and `dotProduct`.
+ * @see https://www.mongodb.com/docs/atlas/atlas-search/field-types/knn-vector/#define-the-index-for-the-fts-field-type-type
+ *
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+final readonly class Store implements VectorStoreInterface
+{
+    /**
+     * @param string $databaseName    The name of the database
+     * @param string $collectionName  The name of the collection
+     * @param string $indexName       The name of the index
+     * @param string $vectorFieldName The name of the field int the index that contains the vector
+     */
+    public function __construct(
+        private Client $client,
+        private LoggerInterface $logger,
+        private string $databaseName,
+        private string $collectionName,
+        private string $indexName,
+        private string $vectorFieldName = 'vector',
+    ) {
+    }
+
+    public function addDocument(Document $document): void
+    {
+        $this->addDocuments([$document]);
+    }
+
+    public function addDocuments(array $documents): void
+    {
+        $collection = $this->client->selectCollection($this->databaseName, $this->collectionName);
+
+        foreach ($documents as $document) {
+            if (!$document->hasVector()) {
+                $this->logger->warning('Document {id} does not have a vector', ['id' => $document->id]);
+            }
+
+            $id = $document->id->toRfc4122();
+
+            $collection->replaceOne(
+                ['id' => $id],
```

Comment:
I noticed that there's no information on an index being required on this field. You should consider using the default `_id` field for this to leverage the index. Note that the identifier does not need to be an ObjectId, but can be any acceptable type for MongoDB fields.

Reply:
You mean like this?

Reply:
Yep, better 👍